src/main/java/org/mitre/synthea/export/FhirStu3.java

```diff
@@ -810,6 +819,284 @@ private static BundleEntryComponent encounterClaim(BundleEntryComponent personEn
     return newEntry(bundle, claimResource);
   }
 
+
+  private static BundleEntryComponent explanationOfBenefit(BundleEntryComponent personEntry,
+                                           Bundle bundle, BundleEntryComponent encounterEntry,
+                                           Person person, org.hl7.fhir.dstu3.model.Claim claim,
+                                                           Encounter encounter) {
+
+    ExplanationOfBenefit eob = new ExplanationOfBenefit();
+    FhirContext fc3 = FhirContext.forDstu3();
+    org.hl7.fhir.dstu3.model.Encounter encounterResource =
+        (org.hl7.fhir.dstu3.model.Encounter) encounterEntry.getResource();
+
+
+
+    // according to CMS guidelines claims have 12 months to be
+    // billed, so we set the billable period to 1 year after
+    // services have ended (the encounter ends).
+    Calendar cal = Calendar.getInstance();
+    cal.setTime(encounterResource.getPeriod().getEnd());
+    cal.add(Calendar.YEAR,1);
+
+    Period billablePeriod = new Period()
+        .setStart(encounterResource
+            .getPeriod()
+            .getEnd())
+        .setEnd(cal.getTime());
+    eob.setBillablePeriod(billablePeriod);
+
+    // cost is hardcoded to be USD in claim so this should be fine as well
+    Money totalCost = new Money();
+    totalCost.setSystem("urn:iso:std:iso:4217");
+    totalCost.setCode("USD");
+    totalCost.setValue(encounter.claim.total());
+    eob.setTotalCost(totalCost);
+
+    // Set References
+    eob.setPatient(claim.getPatient());
+    eob.setOrganization(claim.getOrganization());
+    eob.setProvider(claim.getProvider());
+    eob.setReferral(new Reference("#1"));
+
+    // Hardcoded identifier
+    eob.addIdentifier()
+        .setSystem("https://bluebutton.cms.gov/resources/variables/clm_id")
+        .setValue("23853631084");
```

Comment:
What's the significance of these codes? Should they be hardcoded to a fixed value or generated on a per-eob basis?

Reply:
The `clm_id` is just a unique identifier I'm pretty sure, I've changed the hard coded value with a `UUID.randomUUID().toString()`.  The second part, the `claim group`, I don't know what it is.  It's present in all the example EOB but the url leads to a 404 and I can't find any reference to a `claim group` on the bluebutton website.